framework/web/User.php

```diff
@@ -187,8 +187,16 @@ public function getIdentity($autoRenew = true)
     {
         if ($this->_identity === false) {
             if ($this->enableSession && $autoRenew) {
-                $this->_identity = null;
-                $this->renewAuthStatus();
+                try {
+                    $this->_identity = null;
+                    $this->renewAuthStatus();
+                } catch (\Exception $e) {
+                    $this->_identity = false;
+                    throw $e;
+                } catch (\Throwable $e) {
+                    $this->_identity = false;
```

Comment:
Method return is specified as `IdentityInterface|null`. Returning `false` breaks this contract.